### ISSUE TITLE
Block: Remove dependency on dashboard & design system

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#8327]
## Feature Description

When we originally created the new Web Stories block, we purposely re-used existing components from the dashboard and design system so we could ship this fast and iterate on it later.

That made it easy for us to have the same design as in the dashboard and to display stories properly using the `<CardPreview>`/`<PreviewPage>` component without having to re-invent the wheel.

Now that we're switching to displaying posters instead of actual previews in the dashboard (#5147), it is a good time to cut ties with these packages.

Not only that, it is also a good opportunity to simplify the block's code in other ways.

What I have in mind in total:

1. Remove `ConfigProvider`  
  We can simply access the `window.webStoriesBlockSettings` global
2. Merge `useUserApi` and `useStoryApi` into `useApi`
3. For `useApi`, use `createContext` from `@wordpress/element` instead of from design system.
3. Use UI components from `@wordpress/components` in the story picker instead of components from dashboard and design-system.
4. Use `edit.css` instead of styled components
4. Ensure we display store posters instead of using the full-blown preview components.
5. Simplify API interaction to remove things that we don't need.
  This includes **a lot** of things, including `totalStoriesByStatus` and `reshapeStoryObject`
5. Maybe even ditch `useApi` completely in favor of some simpler callbacks & effects.
6. Maybe: Use `wp.data.select('core').getAuthors()` instead of `fetchAuthors`
7. (Blocked) Maybe: Use `wp.data.select('core').getEntityRecords('postType', 'web-story')` instead of `fetchStories`  
  Currently not possible due to https://core.trac.wordpress.org/ticket/53656 / https://github.com/WordPress/gutenberg/issues/33420 - unless we remove our custom namespace.

Once this is all done there should be no more dependencies to the dashboard and the design system package, nor styled-components, and the block bundle size should be significantly smaller.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by spacedmonkey** [comment]
Wonder if we should do this as the same time https://github.com/google/web-stories-wp/issues/6634

Getting rid of these should do done at the same time 

https://github.com/google/web-stories-wp/blob/9edf5fc1f38933a0bd33ec885ba003524d6a[…]ories-block/block/block-types/selected-stories/selectStories.js

As this is a WordPress / gutenberg context, we should reuse dropdowns and search for gutenberg.

**Comment by swissspidy** [comment]
Yeah I suppose we can, but in a separate PR. Will move it on the board & marked as blocked by this one.

**Comment by swissspidy** [comment]
Assigning to myself since I've been looking into this a bit locally already and have something shareable soon.

**Comment by swissspidy** [comment]
Re-sharing from Slack:

After some initial work to remove dependencies, I got the size of `web-stories-block.js` down from **2.9 MB** (!) to less than **70 KB** (!!!)

**Comment by csossi** [comment]
Both Search Stories and Search By Author have transparent backgrounds in their pulldown-type displays:
![image.png](https://images.zenhubusercontent.com/5ea085e4704e72c0a99c6997/7357bc78-4629-42f8-ac5f-931c3baa1826)
